app/api/v1/endpoints/recommended_challenge.py

```diff
@@ -0,0 +1,115 @@
+# routers/recommended_challenge.py
+
+from fastapi import APIRouter, Depends, HTTPException, status
+from sqlalchemy.ext.asyncio import AsyncSession
+from uuid import UUID
+from typing import List
+
+from app.db.session import get_db
+from app.enum.category import CategoryNameEnum
+
+from app.schemas.base import ResponseBase
+from app.schemas.recommended_challenge import (
+    RecommendedChallengeCreate,
+    RecommendedChallengeUpdate,
+    RecommendedChallengeResponse,
+)
+from app.services.recommended_challenge import (
+    select_recommended_challenge_by_id,
+    select_recommended_challenges,
+    register_recommended_challenge,
+    update_recommended_challenge_by_id,
+    delete_recommended_challenge_by_id,
+)
+
+
+router = APIRouter()
+
+
+# 전체 조회 (user_id + category_name)
+@router.get(
+    "/user/{user_id}/category/{category_name}",
+    response_model=ResponseBase[List[RecommendedChallengeResponse]],
+)
+async def get_recommended_challenges(
+    user_id: UUID,
+    category_name: str,
+    db: AsyncSession = Depends(get_db),
+) -> ResponseBase[List[RecommendedChallengeResponse]]:
+    try:
+        challenges = await select_recommended_challenges(db, user_id, category_name)
+        return ResponseBase(status_code=200, data=challenges)
+    except Exception as e:
+        return ResponseBase(status_code=500, error=str(e))
+
+
+# ✅ 단일 조회 (ID 기준)
+@router.get(
+    "/{challenge_id}",
+    response_model=ResponseBase[RecommendedChallengeResponse],
+)
+async def get_recommended_challenge_by_id(
+    challenge_id: UUID,
+    db: AsyncSession = Depends(get_db),
+) -> ResponseBase[RecommendedChallengeResponse]:
+    try:
+        challenge = await select_recommended_challenge_by_id(db, challenge_id)
```

Comment:
@chaeengg 
실제로 id 가 유일 해서 이것도 잘 조회 될 거긴 하지만 endpoint에서 user_id와 category_id 가 넘어와서 저 같은 경우는 다 넣고 조회 하긴 했습니다! 다만 이건 같이 논의 해보면 좋을 거 같습니다!

@GideokKim @codingorijazz @Ryan-OH 
```suggestion
        challenge = await select_recommended_challenge_by_id(db, user_id, category_id, challenge_id)
```

Reply:
erd상에서 관계가있으니 추가하는게 더 괜찮을 것 같습니다